com.civicrm.hrjobroles/CRM/Hrjobroles/Upgrader.php

```diff
@@ -97,6 +97,91 @@ public function upgrade_1007() {
     return TRUE;
   }
 
+  /**
+   * Migrates existing job role funder to option values
+   *
+   * @return bool
+   */
+  public function upgrade_1008() {
+    $result = civicrm_api3('HrJobRoles', 'get', [
+      'return' => ['funder'],
+    ]);
+    $jobRoles = $result['values'];
+    $funderIds = $this->getFunderIds($jobRoles);
+    $contactFunders = $this->createFunderOptionValues($funderIds);
+    $this->updateJobRoleFunder($jobRoles, $contactFunders);
+
+    return TRUE;
+  }
+
+  /**
+   * Swaps job role funder contact id with option values and updates record
+   *
+   * @param array $jobRoles
+   * @param array $contactFunders
+   */
+  private function updateJobRoleFunder($jobRoles, $contactFunders) {
+    foreach ($jobRoles as $jobRole) {
+      foreach ($contactFunders as $key => $optionValue) {
+        $jobRole['funder'] = str_replace(
+          '|' . $key . '|',
```

Comment:
Just curious, but if you have a list of funders will they all be surrounded by `|` or will it just be the ones in the middle, e.g.

`1|2|3` or `|1|2|3|`?

Reply:
👍 They are surrounded by `|`